src/include/squid/complex.rb

```diff
@@ -49,11 +49,98 @@ def initialize_squid_complex(include_target)
       Yast.include include_target, "squid/store_del.rb"
       Yast.include include_target, "squid/inits.rb"
 
-
       @main_caption = _("Squid")
+      @widget_descr = {}
+      @screens = {}
+
+      load_widgets
+      load_screens
+    end
+
+    def ReallyAbort
+      !Squid.GetModified || Popup.ReallyAbort(true)
+    end
+
+    def PollAbort
+      UI.PollInput == :abort
+    end
+
+    # Read settings dialog
+    #
+    # @return [Symbol] :abort when settings could not be read; :next otherwise
+    def ReadDialog
+      Wizard.RestoreHelp(Ops.get_string(@HELPS, "read", ""))
+      Squid.AbortFunction = fun_ref(method(:PollAbort), "boolean ()")
+
+      return :abort unless Confirm.MustBeRoot
+      return :abort unless PackageSystem.CheckAndInstallPackages(["squid"])
+      return :abort unless Squid.Read
+
+      load_service_widget
+
+      :next
+    end
+
+    # Write settings dialog
+    #
+    # @return [Symbol] :abort if aborted, :next otherwise
+    def WriteDialog
+      help = @HELPS.fetch("write") { "" }
+
+      Wizard.CreateDialog
+      Wizard.RestoreHelp(help)
+      Squid.AbortFunction = fun_ref(method(:PollAbort), "boolean ()")
+      result = Squid.Write
+      Wizard.CloseDialog
+
+      return :next if result
```

Comment:
```suggestion
      result ? :next : :abort
```

Reply:
This time I could not apply directly your suggestion, because it is also needed to remove the next line. However, I'll add it later, in a single commit with your suggestion.

Thank you.